src/main/kotlin/biokotlin/kmer/KmerBigSet.kt

```diff
@@ -0,0 +1,77 @@
+package biokotlin.kmer
+
+import it.unimi.dsi.fastutil.bytes.ByteBigArrays
+import it.unimi.dsi.fastutil.BigArrays
+
+/**
+ * A set of kmers capable of storing up to 134217728 * (2^31 − 1) unique kmers. Includes a count value for each kmer, up to 128.
+ * Recommended use is for dense tables where most possible kmers of length [kmerSize] are present
+ * or where the number of unique kmers exceeds [MAX_VALUE] of Int.
+ */
+class KmerBigSet(kmerSize: Int = 21, bothStrands: Boolean = true, stepSize: Int = 1, keepMinOnly: Boolean = false):
+    AbstractKmerSet(kmerSize, bothStrands, stepSize, keepMinOnly) {
+    override internal var sequenceLength: Long = 0
+    override internal var ambiguousKmers: Long = 0
+
+    internal val arr: Array<ByteArray>
+
+    init {
+        require(kmerSize in 2..28) {"Kmer size must be in the range of 2..28" }
```

Comment:
Kmer size must be between 2 and 28, but sequence size will be 32 bytes (padded if necessary) - is that correct?

Reply:
I don't understand the question. What do you mean by sequence size? Kmers are always longs (64 bits), regardless of the length of the kmer. In KmerBigSet kmers are limited to 28-mers or smaller because of limitations on the size of fastutils BigArrays (though fastutils may change that in the future, in which case we could allow up to 31-mers).

Reply:
Nevermind - I was looking at sequence size defined in Kmer.kt: fun kmer(seq: String) {} - which has a seq len maximum of 32 and then got off on another track.  Ignore this.